internal/controller/websocket.go

```diff
@@ -0,0 +1,74 @@
+package controller
+
+import (
+	"Tradeasy/internal/services/stock_exchange"
+	"encoding/json"
+	"github.com/gin-gonic/gin"
+	"github.com/gorilla/websocket"
+	"log"
+	"time"
+)
+
+//var OrderUpdated = make(chan stock_exchange.OrderResponse)
+
+func StockHandler(c *gin.Context) {
+	var upgrader = websocket.Upgrader{} // use default options
+	// Upgrade our raw HTTP connection to a websocket based one
+	conn, err := upgrader.Upgrade(c.Writer, c.Request, nil)
+	if err != nil {
+		log.Print("Error during connection upgradation:", err)
+		return
+	}
+	defer conn.Close()
+	//write stock details
+	writeEvery(time.Duration(5)*time.Millisecond*1000, conn, stock_exchange.StockWrite)
+}
+func OrderHandler(c *gin.Context) {
+	var upgrader = websocket.Upgrader{} // use default options
+	// Upgrade our raw HTTP connection to a websocket based one
+	conn, err := upgrader.Upgrade(c.Writer, c.Request, nil)
+	if err != nil {
+		log.Print("Error during connection upgradation:", err)
+		return
+	}
+	defer conn.Close()
+
+	/*go func() {
+		for {
+			time.Sleep(5 * time.Second)
+			OrderUpdated <- stock_exchange.OrderResponse{AveragePrice: uint(rand.Intn(10))}
+		}
+	}()*/
+
+	for {
```

Comment:
Where is OrderUpdated defined and what is scope of it?

Reply:
OrderUpdated is defined in services for stock exchange engine

Reply:
Are you initialising OrderUpdated globally? or do you want to keep its scope restricted.